src/MarginTrading.SettingsService/Controllers/ScheduleSettingsController.cs

```diff
@@ -0,0 +1,158 @@
+        public async Task<ScheduleSettingsContract> Insert([FromBody] ScheduleSettingsContract scheduleSetting)
+        {
+            if (string.IsNullOrWhiteSpace(scheduleSetting?.Id))
+            {
+                throw new ArgumentNullException(nameof(scheduleSetting.Id), "scheduleSetting Id must be set");
+            }
+
+            await _scheduleSettingsRepository.InsertAsync(
+                _convertService.Convert<ScheduleSettingsContract, ScheduleSettings>(scheduleSetting));
+
+            await _eventSender.SendSettingsChangedEvent($"{Request.Path}", SettingsChangedSourceType.ScheduleSettings);
+
+            return scheduleSetting;
+        }
+
+        /// <summary>
+        /// Get the schedule setting
+        /// </summary>
+        /// <param name="settingId"></param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("{settingId}")]
+        public async Task<ScheduleSettingsContract> Get(string settingId)
+        {
+            var obj = await _scheduleSettingsRepository.GetAsync(settingId);
+            
+            return _convertService.Convert<ScheduleSettings, ScheduleSettingsContract>(obj);
+        }
+
+        /// <summary>
+        /// Update the schedule setting
+        /// </summary>
+        /// <param name="settingId"></param>
+        /// <param name="scheduleSetting"></param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("{settingId}")]
+        public async Task<ScheduleSettingsContract> Update(string settingId, 
+            [FromBody] ScheduleSettingsContract scheduleSetting)
+        {
+            if (string.IsNullOrWhiteSpace(scheduleSetting?.Id))
+            {
+                throw new ArgumentNullException(nameof(scheduleSetting.Id), "scheduleSetting Id must be set");
+            }
+
+            await _scheduleSettingsRepository.ReplaceAsync(
+                _convertService.Convert<ScheduleSettingsContract, ScheduleSettings>(scheduleSetting));
+
+            await _eventSender.SendSettingsChangedEvent($"{Request.Path}", SettingsChangedSourceType.ScheduleSettings);
+            
+            return scheduleSetting;
+        }
+
+        /// <summary>
+        /// Delete the schedule setting
+        /// </summary>
+        /// <param name="settingId"></param>
+        /// <returns></returns>
+        [HttpDelete]
+        [Route("{settingId}")]
+        public async Task Delete(string settingId)
+        {
+            await _scheduleSettingsRepository.DeleteAsync(settingId);
+
+            await _eventSender.SendSettingsChangedEvent($"{Request.Path}", SettingsChangedSourceType.ScheduleSettings);
+        }
+
+        /// <summary>
+        /// Get the list of compiled schedule settings
+        /// </summary>
+        /// <param name="assetPairIds"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("compiled")]
+        public async Task<List<CompiledScheduleContract>> StateList([FromBody] string[] assetPairIds)

```

Comment:
assetPairIds should be null by default and method will return all available pairs

Reply:
Condition added